lib/services/database/album_data_entry.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:mobileprism/services/database/database_service.dart';
+
+class AlbumDataEntry {
+  late final String uid;
+  late final List<String>? photoUids;
+  late final String title;
+  late final String thumbUid;
+
+  AlbumDataEntry({
+    required this.uid,
+    this.photoUids,
+    required this.title,
+    required this.thumbUid,
+  });
+
+  AlbumDataEntry.fromMap(Map<String, Object?> data) {
+    if (data["uid"] != null &&
+        data["title"] != null &&
+        data["thumb_uid"] != null) {
+      uid = data["uid"]! as String;
+      title = data["title"]! as String;
+      thumbUid = data["thumb_uid"]! as String;
+      if (data["photo_uids"] is List<String>) {
+        photoUids = data["photo_uids"]! as List<String>;
+      } else {
+        photoUids = null;
+      }
+    } else {
+      throw FieldsMissingException();
+    }
+  }
+
+  Map<String, dynamic> toMap() {
+    return {
+      "uid": uid,
+      "photo_uids": photoUids,
+      "title": title,
+      "thumb_uid": thumbUid
+    };
```

Comment:
```dart
  Map<String, dynamic> toJson() => {
        'name': name,
        'email': email,
      };
```
[link](https://docs.flutter.dev/development/data-and-backend/json#serializing-json-inside-model-classes)

Reply:
Basically just rename it right? I agree its better because we have the added functionality, although I think the name is kind of misleading ...

Reply:
Change it to lambda.